app/javascript/components/patient/assessment/steps/SymptomsAssessment.js

```diff
@@ -96,9 +96,18 @@ class SymptomsAssessment extends React.Component {
     this.setState({ selectedBoolSymptomCount: trueBoolSymptoms.length });
   };
 
+  stringifyNumberValues = symptoms => {
+    return symptoms.map(symp => {
+      if (symp.type === 'IntegerSymptom' || symp.type === 'FloatSymptom') {
+        symp.value = symp.value === null ? '' : String(symp.value);
+      }
+      return symp;
+    });
+  };
+
   hasChanges = () => {
     return (
-      !_.isEqual(this.state.reportState.symptoms, this.props.symptoms) ||
+      !_.isEqual(this.state.reportState.symptoms, this.stringifyNumberValues(this.props.symptoms)) ||
```

Comment:
Can you please just add a comment describing why we are stringifying the numbers, since I don't think it's super obvious.

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1255/commits/01dd7c525bb7c38196867719fb1619fe7816e3c8